python/desc/imsim/ImageSimulator.py

```diff
@@ -423,13 +434,19 @@ def __call__(self, gs_objects):
         # Write out the centroid files if they were made.
         gs_interpreter.write_centroid_files()
 
-        # The image for the sensor-visit has been drawn, so delete any
-        # existing checkpoint file if the config says to do so.
+        # The image for the sensor-visit has been drawn, so delete or
+        # move to the archive area any existing checkpoint file if the
+        # config says perform the cleanup.
         if (gs_interpreter.checkpoint_file is not None
                 and os.path.isfile(gs_interpreter.checkpoint_file)
                 and IMAGE_SIMULATOR.config['checkpointing']['cleanup']):
-            os.remove(gs_interpreter.checkpoint_file)
-
+            if IMAGE_SIMULATOR.ckpt_archive_dir is None:
+                os.remove(gs_interpreter.checkpoint_file)
+            else:
+                src_file = gs_interpreter.checkpoint_file
+                dest_file = os.path.join(IMAGE_SIMULATOR.ckpt_archive_dir,
+                                         os.path.basename(src_file))
+                shutil.move(src_file, dest_file)
```

Comment:
In the past, I think I recall we had problems with a small fraction of file operations sometimes not working as they weren't atomic.  Do we need to pay the same attention to such issues here?

Reply:
For the specific use case we added this for, it's not an issue:  if the destination is on the same file system the `shutil.move` should itself be atomic.  For moving to a different fs, the last thing the `shutil.move` would do would be to delete the original file after copying, which would be the last thing our code would do if we tried to make the operation atomic.   In the event of a crash mid-copy to the other fs, there would be a partially written `dest_file`, so I suppose we could copy to a temp file, then rename it and then delete the original.